### PR TITLE
[incubator/solr] Use zookeeper headless service

### PR DESCRIPTION
**Pull request by lwj5**
#### What this PR does / why we need it:
This PR fixes 2 issues:
- Solr startup only checks the first 3 zookeepers for a leader. Let's say we scale zk up to 4 after deployment, and If the leader is now zookeeper-4, solr will not start. We should not hard-code the number.
- ZK client can resolve all IPs from the headless service. We should let it handle the loadbalancing and not use k8s built-in DNS. (I have looked at the zkclient code, and the devs have confirmed this.

#### Special notes for your reviewer:
Solr successfully launches with the default params.
There is a known issue in the web admin interface cloud>zk status.

> Status: red
> Errors:
> We do not have a leader

We can safely ignore this as it's a Solr issue, is purely cosmetic and in the midst of fixing.

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[stable/chart]`)

### CODE REVIEW

File: incubator/solr/templates/statefulset.yaml [hunk 1]
```diff
@@ -63,13 +63,20 @@ spec:
             - |
               COUNTER=0;
               while [  $COUNTER -lt 120 ]; do
-                for i in {{ $vals := . -}}{{ range $i, $e := until ( int .Values.zookeeper.replicaCount ) -}}
-                  "{{- include "solr.zookeeper-name" $vals }}-{{ $i }}.{{ include "solr.zookeeper-service-name" $vals }}" {{ end -}};
-                  do mode=$(echo srvr | nc $i 2181 | grep "Mode");
-                    if [ "$mode" == "Mode: leader" ] || [ "$mode" == "Mode: standalone" ]; then
+                addr=$(nslookup -type=a {{ include "solr.zookeeper-service-name" . }} | grep "Address:" | awk 'NR>1 {print $2}')
```
Comment (paulczar) [inline incubator/solr/templates/statefulset.yaml:66]: I'm concerned the use of `nslookup` might break people that are using custom docker images without nslookup in them.

Comment (lwj5) [reply incubator/solr/templates/statefulset.yaml]: I understand your concern, but this affects only the init container which is using busybox:latest and not the main Solr docker image. I hope this is what you meant.

Besides, there's currently no field to change the init container image in values.yaml and I think people should be "forced" to use an init container with nslookup or remove the init container completely.

Comment (lwj5) [reply incubator/solr/templates/statefulset.yaml:66]: I understand your concern, but this only affects the init container which uses busybox:latest and not the main solr container which I presume that some people would swap for a custom one. I hope this is what you were thinking.

I have tried getent but busybox it does not support it. Besides, there is no field for a user to change the init container image from values.yaml and I think it should be "forced" to use an init container with nslookup or go without an init container at all.

**Pull request merged**